### PR TITLE
Add Heroku hint to Pull Request Template

### PR DESCRIPTION
**Pull request by potatoqualitee**
### Why:

We're asked to check a box for "reviewed my changes in staging", but for those of us who are external to this project, we may not know where to look. I didn't until it was highlighted later on during my previous PR review.

### What's being changed:

Adds a hint as to where to look.

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -22,6 +22,6 @@ Thanks again!
 
 ### Check off the following:
 - [ ] All of the tests are passing.
-- [ ] I have reviewed my changes in staging.
+- [ ] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in the resulting Pull Request).
```
Comment (heiskr) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
- [ ] I have reviewed my changes in staging. Look for the **deploy-to-heroku** link in the resulting pull request.
```

**Review by heiskr — APPROVED** [review]
🚢

**Comment by alialthaib** [comment]
Go

**Comment by alialthaib** [comment]
Good

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (janiceilene) [reply .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
- [ ] I have reviewed my changes in staging. Look for the **deploy-to-heroku** link in your pull request, then click **View deployment**.
```
💝 @potatoqualitee I had this update on my to do list and was just not getting to it. Thank you so much for opening a PR! I used @heiskr's suggestion and expanded it to let folks know what button to push. Do you think it's clear?

Comment (potatoqualitee) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Yes! looks great, @janiceilene ! 🎉 My only concern with the wording is that I goes from first person to third person ("I have" to "Look for"). The parens helped the reader see that it was a hint for them. But it's not a show-stopper.

Comment (potatoqualitee) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Perhaps `I have looked for the  **deploy-to-heroku** link in my pull request, then clicked **View deployment** and reviewed my changes in staging` though that feels clunky.

Comment (janiceilene) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Ooh, good call 🙌  I just updated my suggestion to include the parenthesis. If that looks good to you, feel free to commit the suggestion and we'll get this merged in soon!

Comment (janiceilene) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: This one slipped through the cracks! Sorry about that @potatoqualitee 💛 How do you feel about the language in my suggestion? If you want to commit that suggestion (or change it in some way), I'll get this merged in once you're ready!

**Comment by janiceilene** [comment]
This is not stale! I'm going to commit the suggestion I made and get it merged in ⚡

**Pull request merged**